tests/robustness/validate/patch_history_test.go

```diff
@@ -233,12 +265,29 @@ func TestPatchHistory(t *testing.T) {
 				deleteRequest("key"),
 				putRequest("key", "value"),
 			},
-			watchOperations: watchResponse(3, deleteEvent("key", 2)),
 			expectedRemainingOperations: []porcupine.Operation{
-				{Return: 399, Output: model.MaybeEtcdResponse{Persisted: true, PersistedRevision: 2}},
+				{Return: 399, Output: model.MaybeEtcdResponse{Persisted: true}},
 				{Return: 400, Output: putResponse(0, model.EtcdOperationResult{})},
 			},
 		},
+		{
+			name: "failed delete remains if there is a matching persisted request, lack of uniqueness of this operation prevents patching based on following operation",
+			historyFunc: func(h *model.AppendableHistory) {
+				h.AppendDelete("key", 100, 200, nil, errors.New("failed"))
+				h.AppendPut("key", "value", 300, 400, &clientv3.PutResponse{}, nil)
+				h.AppendDelete("key", 500, 600, &clientv3.DeleteResponse{}, nil)
+			},
+			persistedRequest: []model.EtcdRequest{
+				deleteRequest("key"),
+				putRequest("key", "value"),
+				deleteRequest("key"),
+			},
+			expectedRemainingOperations: []porcupine.Operation{
+				{Return: infinite, Output: model.MaybeEtcdResponse{Error: "failed"}},
+				{Return: 400, Output: putResponse(0, model.EtcdOperationResult{})},
+				{Return: 600, Output: putResponse(0, model.EtcdOperationResult{})},
```

Comment:
Nit: this threw me off a bit because line 278 has a delete operation, but this line has `putResponse`.  It seems we just need the same response as the one for time=400 where the accepted PUT happened.  Maybe a comment would help.

Reply:
`putResponse` threw me off to. Will rename it in followup PR.